make_site.py

```diff
@@ -170,36 +172,40 @@ class DocDecl:
     # is simply a link to the right revision of the mathlib source code.
     src_link: str
 
+# keep public fields in sync with `scripts/yaml_check.py` in mathlib4 repo
 @dataclass
 class HundredTheorem:
     number: str
     title: str
-    decl: Optional[str] = None
     decls: Optional[List[str]] = None
-    # The HTML source code for the generated documentation entry
-    # for the declaration(s) in |decl| or |decls|.
-    doc_decls: Optional[List[DocDecl]] = None
     author: Optional[str] = None
+    date: Optional[str] = None
     links: Optional[Mapping[str, str]] = None
     note: Optional[str] = None
+    # The following is a private field populated with data from `header-data.json` and
+    # should NOT be present in `100.yaml` or `scripts/yaml_check.py`
+    # The HTML source code for the generated documentation entry
+    # for the declaration(s) in |decl| or |decls|.
+    _doc_decls: Optional[List[DocDecl]] = None
 
+# keep public fields in sync with `scripts/yaml_check.py` in mathlib4 repo
 @dataclass
 class ThousandPlusTheorem:
     # Wikidata identifier (the letter Q followed by a string as digits),
     # optionally followed by a letter (such as "A", "B" or "X" for disambiguation).
     # "Q1008566" and "Q4724004A" are valid identifiers, for example.
     wikidata: str
     title: str
-    decl: Optional[str] = None
     decls: Optional[List[str]] = None
-    # The HTML source code for the generated documentation entry
-    # for the declaration(s) in |decl| or |decls|.
-    doc_decls: Optional[List[DocDecl]] = None
     author: Optional[str] = None
     date: Optional[str] = None
-    url: Optional[str] = None
+    links: Optional[Mapping[str, str]] = None
```

Comment:
To be honest, I like the `url` field for `1000.yaml`. (This also matches the upstream specification, which just allows putting one URL.) I'd rather not change this aspect.
You *are* right that 1000.html was previously wrong, but I'd propose to unify these conventions at a different level.

Reply:
I'm actually a bit surprised that you prefer `url`, given that there's already an entry in `1000.yaml` where it seems having a `links`-style list of pairs would be more flexible: https://github.com/leanprover-community/mathlib4/blob/62fd5798b4a61e5cbeee6e32562dcb4cbd66e0b1/docs/1000.yaml#L145

However, as I said above, I'm happy to go along with you on this.

Reply:
If we have a single `url` field upstream, but allow `links` here, there will be pain converting between these two. I'm not convinced that is a good strategy. That said: you *are* right that Gödel's incompleteness theorem should be split in two. In other words, one should file a PR upstream, splitting this into two results. I'll add that to my TODO list.

Reply:
@fpvandoorn What do you think about this question? Should one have a single URL upstream, or allow multiple URLs? (Case in point, Gödel's incompleteness theorem.) I think having a single URL is nice --- and would thus propose splitting the upstream entry instead. What do you think?

(Not urgent: we can discuss this in the new year.)